src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonSyncIntTest.kt

```diff
@@ -1,36 +1,439 @@
+import uk.gov.justice.hmpps.sqs.countAllMessagesOnQueue
+import java.time.LocalDateTime
+import java.time.ZoneId
 
 class PrisonPersonSyncIntTest : SqsIntegrationTestBase() {
 
-  @Test
-  @DisplayName("Temporary test to prove the test listeners are wired up - to be replaced with some proper sync tests")
-  fun `can consume events`(output: CapturedOutput) {
-    awsSqsPrisonPersonOffenderEventsClient.sendMessage(
-      prisonPersonQueueOffenderEventsUrl,
-      physicalAttributesChangedEvent(),
-    )
+  @Autowired
+  private lateinit var nomisMockServer: PrisonPersonNomisApiMockServer
 
-    await untilAsserted {
-      assertThat(output).contains("Received physical attributes changed event")
+  private val now = LocalDateTime.now()
+  private val yesterday = now.minusDays(1)
+
+  @Nested
+  @DisplayName("OFFENDER_PHYSICAL_ATTRIBUTES-CHANGED")
+  inner class PhysicalAttributesChanged {
+    @Nested
+    inner class SingleBooking {
+
+      @Test
+      fun `should sync new physical attributes`() {
+        nomisMockServer.stubGetPhysicalAttributes(
+          aPrisonerPhysicalAttributesResponse(
+            bookings = listOf(
+              aBookingResponse(
+                startDateTime = "$yesterday",
+                endDateTime = null,
+                physicalAttributes = listOf(
+                  aPhysicalAttributesResponse(
+                    heightCentimetres = 180,
+                    weightKilograms = 80,
+                    modifiedDateTime = "$now",
+                    modifiedBy = "A_USER",
+                  ),
+                ),
+              ),
+            ),
+          ),
+        )
+        dpsPrisonPersonServer.stubSyncPrisonPerson(aPhysicalAttributesHistoryDto())
+
+        sendPhysicalAttributesChangedEvent()
+
+        verifyResults(
+          dpsUpdates = mapOf(
+            "height" to equalTo("180"),
+            "weight" to equalTo("80"),
+            "appliesFrom" to equalTo("${yesterday.toZoned()}"),
+            "appliesTo" to absent(),
+            "createdAt" to equalTo("${now.toZoned()}"),
+            "createdBy" to equalTo("A_USER"),
+          ),
+        )
+      }
+
+      @Test
+      fun `should ignore a new empty booking (when a new prisoner is created)`() {
+        nomisMockServer.stubGetPhysicalAttributes(
+          aPrisonerPhysicalAttributesResponse(
+            bookings = listOf(
+              aBookingResponse(
+                endDateTime = null,
+                physicalAttributes = listOf(
+                  aPhysicalAttributesResponse(
+                    heightCentimetres = null,
+                    weightKilograms = null,
+                    modifiedDateTime = null,
+                  ),
+                ),
+              ),
+            ),
+          ),
+        )
+
+        sendPhysicalAttributesChangedEvent()
+
+        verifyResults(type = "ignored", reason = "New physical attributes are empty")
```

Comment:
Question; are physical attributes not copied by NOMIS for new bookings (like it does for Alerts)?

Reply:
They are yes. This test is for a brand new booking - if you go into the physical info NOMIS screen it creates this "empty" record which makes no sense to synchronise.